dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -118,6 +118,13 @@ public async Task<Plan> CreatePlanAsync(string goal)
             plan = new Plan(goal);
         }
 
+        // Planner should not suggest server-url as a parameter.
+        // Server URL should be be parsed from OpenAPI spec or provided by user
+        if (planData.Plan.Parameters.ContainsKey("server-url"))
```

Comment:
I think the better fix would be to update the functions for specifying functions to include their default values. See `defaultValue` usage in dotnet\src\Extensions\Planning.SequentialPlanner\FunctionViewExtensions.cs

Reply:
Things like Jira won't have a default url directly associated with the specification. We need to be able to provide this at runtime.

Reply:
@teresaqhoang  - will this change prevent proper usage of default values?

Reply:
No it won't, the default value lives in the plan's kernel's context. This is just removing the server-url parameter from the plan's suggested parameters collection. The following lines 129-141 update the planner's kernel's context with the plan's suggested parameters (which is why server-url was being overridden before).

Reply:
I would do that on plan consumption or creation. Who supplies the server-url? Can it be part of the planning prompt? Or set the value in the context you give to the plan invocation to override the plan parameters.

Reply:
Also this isnt fool proof, if the planner doesnt formulate the ask with the server-url as an input, it can fail much earlier, with a "Plan parsing error, invalid JSON". This happened a bunch for jira because the openapi spec cant provide a usable default value for the server url. And atleast right now the server url we ask users to enter is not being passed along to the backend. I had to hard code the variable to get consistent results.

Reply:
The "Plan parsing error, invalid JSON" error is due to a different bug in ActivePlanner -- the planner tries to return 2 plans so the resulting JSON is invalid. @amsacha if you come across this error again, can you confirm? You can do this by putting a breakpoint right after the CreatePlanAsync function is called in ChatSkill and inspecting the plan (line 214 in the screenshot)
<img width="869" alt="Screenshot 2023-05-03 at 8 05 19 PM" src="https://user-images.githubusercontent.com/125500434/236104409-0b45caf7-a2b5-4a56-aac8-1368847583a4.png">

I don't think hardcoding the value is the best method, it's passed in the context variables now, so maybe we can add a wrapper in Plan.cs when evaluating namedParameters?